app/src/main/java/uk/gov/android/authentication/AppAuthSession.kt

```diff
@@ -20,85 +15,27 @@ class AppAuthSession(
-            )
-            persistentSessionId?.let {
-                additionalParameters["govuk_signin_session_id"] = it
-            }
-
-            val builder =
-                AuthorizationRequest.Builder(
-                    serviceConfig,
-                    clientId,
-                    responseType.value,
-                    redirectUri
-                )
-                    .setScopes(scopes.map { it.value })
-                    .setUiLocales(locale.value)
-                    .setNonce(nonce)
-                    .setAdditionalParameters(additionalParameters)
-
-            val authRequest = builder.build()
-
-            val authIntent = authService.getAuthorizationRequestIntent(authRequest)
-            launcher.launch(authIntent)
-        }
+        val intent = authService.getAuthorizationRequestIntent(configuration.createRequest())
+        launcher.launch(intent)
     }
 
     override fun finalise(
         intent: Intent,
         callback: (tokens: TokenResponse) -> Unit
     ) {
-        val authorizationResponse = AuthorizationResponse.fromIntent(intent)
-
-        if (authorizationResponse == null) {
-            val exception = AuthorizationException.fromIntent(intent)
-
-            throw AuthenticationError(
-                exception?.message ?: "Auth response was null",
-                AuthenticationError.ErrorType.OAUTH
-            )
-        }
-
-        val exchangeRequest = authorizationResponse.createTokenExchangeRequest()
-
+        val authResponse = AuthorizationResponse.fromIntent(intent)
+        val request = authResponse?.createTokenExchangeRequest() ?: throw AuthenticationError.from(intent)
         authService.performTokenRequest(
-            exchangeRequest
+            request
         ) { response, exception ->
-            if (response == null) {
-                throw AuthenticationError(
-                    exception?.message ?: "Failed token request",
-                    AuthenticationError.ErrorType.OAUTH
-                )
-            }
-
-            callback(createFromAppAuthResponse(response))
+            callback(
+                response?.toTokenResponse() ?: throw AuthenticationError.from(exception)
+            )
         }
     }
 
-    private fun createFromAppAuthResponse(
-        response: net.openid.appauth.TokenResponse
-    ): TokenResponse {
-        return TokenResponse(
-            tokenType = requireNotNull(response.tokenType) { "token type must not be empty" },
-            accessToken =
-            requireNotNull(response.accessToken) { "access token must not be empty" },
-            accessTokenExpirationTime =
-            requireNotNull(response.accessTokenExpirationTime) {
-                "Token expiry must not be empty"
-            },
-            idToken = response.idToken,
-            refreshToken = response.refreshToken
-        )
+    companion object {
+        @Suppress("unused")
+        const val REQUEST_CODE_AUTH = 418
```

Comment:
Don't think we need this anymore... Think it was a remontant from using `Activity.launchActivityForResult()`

Reply:
Removed it